ui/app/src/components/PathNavigatorTree/PathNavigatorTreeItem.tsx

```diff
@@ -0,0 +1,143 @@
+import { FormattedMessage } from 'react-intl';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import ItemDisplay from '../ItemDisplay';
+import Tooltip from '@material-ui/core/Tooltip';
+import IconButton from '@material-ui/core/IconButton';
+import MoreVertRoundedIcon from '@material-ui/icons/MoreVertRounded';
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    root: {
+      '&:focus > $content $labelContainer': {
+        backgroundColor: 'transparent'
+      }
+    },
+    content: {},
+    labelContainer: {
+      display: 'flex',
+      paddingLeft: 0,
+      height: '26px',
+      '& .options': {
+        display: 'none'
+      },
+      '&:hover': {
+        backgroundColor: `${theme.palette.action.hover} !important`,
+        '& .options': {
+          display: 'block'
+        }
+      }
+    },
+    iconContainer: {
+      width: '26px',
+      marginRight: 0,
+      '& svg': {
+        fontSize: '26px',
+        color: theme.palette.text.secondary
+      }
+    },
+    loading: {
+      display: 'flex',
+      alignItems: 'center',
+      height: '26px',
+      marginLeft: '10px',
+      '& span': {
+        marginLeft: '10px'
+      }
+    }
+  })
+);
+
+interface PathNavigatorTreeItemProps {
+  node: TreeNode;
+  itemsByPath: LookupTable<DetailedItem>;
+  onLabelClick(path: string): void;
+  onIconClick(path: string): void;
+  onOpenItemMenu(element: Element, path: string): void;
+}
+
+export default function PathNavigatorTreeItem(props: PathNavigatorTreeItemProps) {
+  const { node, itemsByPath, onLabelClick, onIconClick, onOpenItemMenu } = props;
+  const classes = useStyles();
+  return node.id === 'loading' ? (
+    <div className={classes.loading}>
+      <CircularProgress size={14} />
+      <Typography variant="caption" color="textSecondary">
+        <FormattedMessage id="words.loading" defaultMessage="Loading" />
+      </Typography>
+    </div>
+  ) : (
+    <TreeItem
+      key={node.id}
+      nodeId={node.id}
+      onLabelClick={() => onLabelClick(node.id)}
+      onIconClick={() => onIconClick(node.id)}
+      label={
+        <>
+          <ItemDisplay
+            styles={{ root: { maxWidth: '100%', flexGrow: 1 } }}
+            item={itemsByPath[node.id]}
+            showPublishingTarget={true}
+            showWorkflowState={true}
+            labelTypographyProps={{ variant: 'body2' }}
+          />
+          <section className="options">
+            <Tooltip title={<FormattedMessage id="words.options" defaultMessage="Options" />}>
+              <IconButton
+                size="small"
+                onClick={(e) => {
+                  e.preventDefault();
+                  e.stopPropagation();
+                  onOpenItemMenu(e.currentTarget, node.id);
+                }}
+              >
+                <MoreVertRoundedIcon fontSize="small" />
+              </IconButton>
+            </Tooltip>
+          </section>
```

Comment:
Pull & checkout the tweaks on PathNavigator to avoid tooltip jumps/layout-issues